content/doc/book/system-administration/diagnosing-build-failures.adoc

```diff
@@ -0,0 +1,93 @@
+---
+layout: section
+---
+ifdef::backend-html5[]
+ifndef::env-github[:imagesdir: ../../resources/managing]
+:notitle:
+:description:
+:author:
+:email: jenkinsci-users@googlegroups.com
+:sectanchors:
+:toc: left
+endif::[]
+
+= Diagnosing Build Failures
+
+
+== Builds running out of memory?
+
+As your project grows, and you use new tools to either build or analyze
+your code, you will inevitably exceed the memory settings which your JVM
+provides by default. This is especially true on 64 bit JVM's since they
+double the size of the reference pointer. This page aims to show you how
+to increase the memory available to your build process.
+
+
+=== Heap or Permgen?
+
+There are two OutOfMemoryErrors which people usually encounter. The
+first is related to heap space:
+`+java.lang.OutOfMemoryError: Heap space+` When you see this, you need
+to increase the maximum heap space. You can do this by adding the
+following to your JVM arguments `+-Xmx200m+` where you replace the
+number 200 with the new heap size in megabytes.
+
+The second is related to PermGen:
+`+java.lang.OutOfMemoryError: PermGen space+`. When you see this, you
+need to increase the maximum Permanent Generation space, which is used
+for things like class files and interned strings. You can do this by
+adding the following to your JVM arguments
+`+-XX:MaxPermSize=128m+` where you replace the number 128 with the new
+PermGen size in megabytes.
+
+=== Various Build Tools
+
+This section demonstrates how to set these JVM switches depending on the
+kind of build you are running.
+
+==== Maven2/3 Project Type
```

Comment:
https://www.cloudbees.com/blog/painless-maven-builds-jenkins calls it the "Maven 2/3 project type" (with a space between "Maven" and "2".  Seems safe to do it here as well.
```suggestion
==== Maven 2/3 Project Type
```